java.io.FileNotFoundException:  ..../keycloak-17.0.0/model/jpa/target/test-classes/log4j.properties

### Describe the bug

Following the document https://github.com/keycloak/keycloak/blob/main/docs/building.md to build keycloak, the build is failed due to the missing of log4j.properties.

[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ keycloak-model-jpa ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.keycloak.connections.jpa.util.JpaUtilsTest
log4j:ERROR Could not read configuration file from URL [file:<path-ro-keycloak>/keycloak/template/keycloak-17.0.0/model/jpa/target/test-classes/log4j.properties].
java.io.FileNotFoundException:<path-ro-keycloak>/keycloak/template/keycloak-17.0.0/model/jpa/target/test-classes/log4j.properties (No such file or directory)

$ java -version
openjdk version "17.0.1" 2021-10-19
OpenJDK Runtime Environment (build 17.0.1+12-Ubuntu-120.04)
OpenJDK 64-Bit Server VM (build 17.0.1+12-Ubuntu-120.04, mixed mode, sharing)

$ mvn -version
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
....

I tried to build keycloak so that I can translate the messages to Traditional Chinese later. Please help. Thanks.

### Version

17

### Expected behavior

build successfully?

### Actual behavior

[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M5:test (default-test) @ keycloak-model-jpa ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.keycloak.connections.jpa.util.JpaUtilsTest
log4j:ERROR Could not read configuration file from URL [file:<path-ro-keycloak>/keycloak/template/keycloak-17.0.0/model/jpa/target/test-classes/log4j.properties].
java.io.FileNotFoundException:<path-ro-keycloak>/keycloak/template/keycloak-17.0.0/model/jpa/target/test-classes/log4j.properties (No such file or directory)


### How to Reproduce?

$ wget https://github.com/keycloak/keycloak/archive/refs/tags/17.0.0.tar.gz
$ tar xfz 17.0.0.tar.gz 
$ cd keycloak-17.0.0/
$ mvn clean install


### Anything else?

_No response_

I ran the build with JDK 17 and received the following error message at the end: 

``` 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:22 min
[INFO] Finished at: 2022-03-10T09:55:18+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.hibernate.orm.tooling:hibernate-enhance-maven-plugin:5.3.20.Final:enhance (default) on project keycloak-model-map-jpa: Unable to enhance class: JpaRoleAttributeEntity.class: Failed to enhance class org.keycloak.models.map.storage.jpa.role.entity.JpaRoleAttributeEntity: Unsupported class file major version 61 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :keycloak-model-map-jpa
```

The error message you reported is also there somewhere in the log, but the log you reported is not the reason for the build to fail. If I understand the log correctly, the current hibernate plugin we're using doesn't support JDK 17, yet. There is already an issue for this here #9993

I will use this issue to get rid of the error message in the log about log4j; there'll be PR for this shortly.

--- a/model/jpa/pom.xml
+++ b/model/jpa/pom.xml
@@ -136,7 +136,8 @@
                         <keycloak.connectionsJpa.default.user>${keycloak.connectionsJpa.user}</keycloak.connectionsJpa.default.user>
                         <keycloak.connectionsJpa.default.password>${keycloak.connectionsJpa.password}</keycloak.connectionsJpa.default.password>
                         <keycloak.connectionsJpa.default.url>${keycloak.connectionsJpa.url}</keycloak.connectionsJpa.default.url>
-                        <log4j.configuration>file:${project.build.directory}/test-classes/log4j.properties</log4j.configuration> <!-- for the logging to properly work with tests in the 'other' module -->
+                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        <org.jboss.logging.provider>log4j</org.jboss.logging.provider>
                     </systemPropertyVariables>
                 </configuration>
             </plugin>

--- a/model/jpa/src/test/resources/log4j.properties
+++ b/model/jpa/src/test/resources/log4j.properties
@@ -1,0 +1,5 @@
+log4j.rootLogger=info, stdout
+
+log4j.appender.stdout=org.apache.log4j.ConsoleAppender
+log4j.appender.stdout.layout=org.apache.log4j.PatternLayout
+log4j.appender.stdout.layout.ConversionPattern=%d{HH:mm:ss,SSS} %-5p %t [%c] %m%n

